### PR TITLE
layers: GH384 Fix to correctly recognize immutable samplers

### PR DESCRIPTION
**Pull request by tobine**
Actual fix this time. If a binding is immutable samplers
skip over it when verifying descriptor bindings.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2668,16 +2668,17 @@ static bool validate_and_update_drawtime_descriptor_state(
...
                 // TODO : Would be better to store set# that set is bound to so we can report set.binding[index] not updated
                 // For immutable sampler w/o combined image, don't need to update
-                if ((set_node->pLayout->createInfo.pBindings[i].descriptorType == VK_DESCRIPTOR_TYPE_SAMPLER) &&
-                    (set_node->pLayout->createInfo.pBindings[i].descriptorCount != 0) &&
-                    (set_node->pLayout->createInfo.pBindings[i].pImmutableSamplers)) {
-                    // Nothing to do here
+                if ((set_node->pLayout->createInfo.pBindings[binding_index].descriptorType == VK_DESCRIPTOR_TYPE_SAMPLER) &&
+                    (set_node->pLayout->createInfo.pBindings[binding_index].descriptorCount != 0) &&
+                    (set_node->pLayout->createInfo.pBindings[binding_index].pImmutableSamplers)) {
+                    i = endIdx; // Advance to end of this binding, no more work for immutable samplers
```
Comment (lentinem) [inline layers/core_validation.cpp]: This is the same as a break which would be a bit cleaner but this check also doesn't depend on i so can't we do this outside the loop and only do the loop if false?

**Comment by tobine** [comment]
Yeah, I like that better, Michael. Pushed updated.

**Comment by lentinem** [comment]
+1

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**